pkg/helm/client.go

```diff
@@ -76,36 +41,44 @@ func newClient(opts HelmOptions) (*HelmClient, error) {
 		}
 		kversion = sv
 	}
-	regcli, err := registry.NewClient(registryOpts...)
-	if err != nil {
-		return nil, fmt.Errorf("create registry client: %w", err)
-	}
-	if opts.RESTClientGetter == nil {
-		cfgFlags := &genericclioptions.ConfigFlags{}
-		if opts.KubeConfig != "" {
-			cfgFlags.KubeConfig = &opts.KubeConfig
-		}
-		opts.RESTClientGetter = cfgFlags
-	}
+
 	return &HelmClient{
-		tmpdir:           tmpdir,
-		kversion:         kversion,
-		restClientGetter: opts.RESTClientGetter,
-		regcli:           regcli,
-		logFn:            opts.LogFn,
-		airgapPath:       opts.AirgapPath,
+		helmPath:              helmPath,
+		executor:              newBinaryExecutor(helmPath, opts.LogFn),
+		tmpdir:                tmpdir,
+		kversion:              kversion,
+		kubernetesEnvSettings: opts.KubernetesEnvSettings,
+		airgapPath:            opts.AirgapPath,
+		repositories:          []*repo.Entry{},
 	}, nil
 }
 
+func getHelmPath(opts HelmOptions) (string, error) {
+	if opts.HelmPath != "" {
+		// used in operations that do not have the helm binary embedded (i.e. upgrade via operator, etc..)
+		return opts.HelmPath, nil
+	}
+	if hp := os.Getenv("HELM_BINARY_PATH"); hp != "" {
+		// used in tests so that we don't pass the helm binary path all over the place
+		return hp, nil
+	}
+	hp, err := goods.Binary("helm")
+	if err != nil {
+		return "", fmt.Errorf("get embedded helm binary: %w", err)
+	}
+	return hp, nil
+}
+
 type HelmOptions struct {
-	KubeConfig       string
-	RESTClientGetter genericclioptions.RESTClientGetter
-	K8sVersion       string
-	AirgapPath       string
-	Writer           io.Writer
-	LogFn            action.DebugLog
+	HelmPath              string
+	KubernetesEnvSettings *helmcli.EnvSettings
+	K8sVersion            string
```

Comment:
is K8sVersion no longer used? Is this going to be an issue for things like buildtools or rendering before install k0s? If not can you remove it?

Reply:
it is used